### DISCUSSION TITLE
Problems in stargate pod startup

### DISCUSSION THREAD

**Discussion by andreasgeisslerdt** [stargate/stargate#2568] [Help] [answered]
I deployed stargate using the k8ssandra-operator.
But the stargate is not starting up due to issues starting up with the problem of starting the auth server:

```
INFO  [main] 2023-05-08 11:34:27,497 AbstractServerFactory.java:750 - Starting AuthApiServer
ERROR [main] 2023-05-08 11:34:27,858 ServerCommand.java:54 - Unable to start server, shutting down
java.lang.RuntimeException: java.io.IOException: Failed to bind to 0.0.0.0/0.0.0.0:8081
...
Caused by: java.net.BindException: Address already in use
...
```
Does someone knows the reason for this issue ?

Thanks and best regards
Andreas

kassandracluster.yaml:
```
apiVersion: k8ssandra.io/v1alpha1
kind: K8ssandraCluster
...
cassandra:
    serverVersion: 4.0.1
    storageConfig:
      cassandraDataVolumeClaimSpec:
        storageClassName: cinder-os
        accessModes:
          - ReadWriteOnce
        resources:
          requests:
            storage: 5Gi
    config:
      jvmOptions:
        heapSize: 512M
    networking:
      hostNetwork: true
    datacenters:
      - metadata:
         name: dc1
         #        k8sContext: dc1
        size: 3
  stargate:
    size: 1
```

**Comment by ivansenic** [comment]
Well the problem is that port 8081 is already taken on the pod, but I have no idea why.. I think it would be best to ask directly on the https://github.com/k8ssandra/k8ssandra-operator.. 

Or maybe @adejanovski or @olim7t know the answer out-of-the-box?

**Reply by adejanovski** [reply]
@andreasgeisslerdt, you should try removing the networking part of your manifest which enables host networking.
This should only be used in specific cases (testing multicluster in kind for example) and will have your pods use the host IP address instead of a different one. That can lead to conflicts around ports.

**Reply by andreasgeisslerdt** [reply]
Indeed I see in the stargate deployment the setting:
`      hostNetwork: true`
and in the pod:
```
    Image:          artifactory.devops.telekom.de/hub.docker.com/stargateio/stargate-4_0:v1.0.76
    Image ID:       artifactory.devops.telekom.de/hub.docker.com/stargateio/stargate-4_0@sha256:f48efe1c1710a5e2ff40af3e1c98a0967269273ccd1614c1faaeab85d8f2b648
    Ports:          8080/TCP, 8081/TCP, 8082/TCP, 8084/TCP, 8085/TCP, 8090/TCP, 9042/TCP, 8609/TCP, 7000/TCP, 7001/TCP
    Host Ports:     8080/TCP, 8081/TCP, 8082/TCP, 8084/TCP, 8085/TCP, 8090/TCP, 9042/TCP, 8609/TCP, 7000/TCP, 7001/TCP
```
I will have to see, how I can disable the hostNetwork setting in the k8ssandraCluster settings.
Thanks a lot for the hint !!!

**Reply by adejanovski** [reply, marked as answer]
Sure, just remove this block from your manifest:
```
    networking:
      hostNetwork: true
```

**Reply by andreasgeisslerdt** [reply]
Thanks, I just tested it !!!

**Reply by elizabethanderson2** [reply]
i dont know

**Reply by nutter4x45** [reply]
wen stargate v2&&
